app/services/foreman/parameters/validator.rb

```diff
@@ -43,6 +43,23 @@ def validate_list
         true
       end
 
+      def validate_hash
```

Comment:
It looks like you've removed code that calls this, but not these methods.  Are they unused now?